src/pkgdev/cli.py

```diff
@@ -11,3 +20,106 @@ def main(self):
+
+    def __call__(self, parser, namespace, values, option_string=None):
+        if values.lower() in ('false', 'no', 'n'):
+            values = False
+        setattr(namespace, self.dest, values)
+
+
+class ConfigParser(configparser.ConfigParser):
+    """ConfigParser with case-sensitive keys (default forces lowercase)."""
+
+    def optionxform(self, option):
+        return option
+
+
+class ConfigFileParser:
+    """Argument parser that supports loading settings from specified config files."""
+
+    default_configs = (const.SYSTEM_CONF_FILE, const.USER_CONF_FILE)
+
+    def __init__(self, parser, configs=(), **kwargs):
+        super().__init__(**kwargs)
+        self.parser = parser
+        self.configs = OrderedSet(configs)
+
+    @jit_attr_none
+    def config(self):
+        return self.parse_config()
+
+    def parse_config(self, configs=()):
+        """Parse given config files."""
+        configs = configs if configs else self.configs
+        config = ConfigParser(default_section=None)
+        try:
+            for f in configs:
+                config.read(f)
+        except configparser.ParsingError as e:
+            self.parser.error(f'parsing config file failed: {e}')
+        return config
+
+    def parse_config_sections(self, namespace, sections):
+        """Parse options from a given iterable of config section names."""
+        for section in (x for x in sections if x in self.config):
+            config_args = [f'--{k}={v}' if v else f'--{k}' for k, v in self.config.items(section)]
+            namespace, _ = self.parser.parse_known_optionals(config_args, namespace)
+        return namespace
+
+    def parse_config_options(self, namespace, configs=()):
+        """Parse options from config if they exist."""
+        configs = list(filter(os.path.isfile, configs))
+        if not configs:
+            return namespace
+
+        self.configs.update(configs)
+        # reset jit attr to force reparse
+        self._config = None
+
+        # load default options
+        namespace = self.parse_config_sections(namespace, ['DEFAULT'])
+
+        return namespace
+
+class ArgumentParser(arghparse.ArgumentParser):
+    """Parse all known arguments, from command line and config file."""
+
+    def __init__(self, parents=(), **kwargs):
+        self.config_argparser = arghparse.ArgumentParser(suppress=True)
+        config_options = self.config_argparser.add_argument_group('config options')
+        config_options.add_argument(
+            '--config', action=ConfigArg, dest='config_file',
+            help='use custom pkgdev settings file',
+            docs="""
+                Load custom pkgdev scan settings from a given file.
+
+                Note that custom user settings override all other system and repo-level
+                settings.
+
+                It's also possible to disable all types of settings loading by
+                specifying an argument of 'false' or 'no'.
```

Comment:
I don't know if this is a common feature of `--config` flags, or flags accepting arguments in pkgcheck, but this feels like the flag is doing two separate things. If I'm nasty, I name my config file `no` then it's not gonna read the file called "no", but just not load any configuration, right? (As I read it `false`, `no` and `n` are treated as "do not load configs", right?)

Is there a case for a separate flag to do this? `--no-load-configs` or something?

Reply:
From fast scan in the pkgcore stack, it looks like only `pkgcheck scan` uses this system (on which this implementation is super heavily based).
I can do it using `--no-config` - maybe it is even a better idea, but then I need help and documentation strings for it 🥲

Reply:
If there's precedence for this, it might make more sense to keep it consistent. But if you'd prefer to add a `--no-config` flag, I'm happy to write some doc strings.

Reply:
After various considerations, I prefer it to match pkgcheck's one, meaning the current one (with special value `no` and `false`)

Reply:
OK :+1: